File: src/perceptron/Teacher.java
package perceptron;

import generic.NumericDataSet;

public class Teacher {

    public static void teach(Perceptron perceptron, NumericDataSet numericDataSet, String classValue) {
        boolean complete = false;
        int counter = 1;
        while (!complete) {
            complete = true;
            for (int i = 0; i < numericDataSet.getAllDataPoints().size(); i++) {
                int d = classValue.equals(numericDataSet.getAllDecisions().get(i)) ? 1 : 0;
                double y = perceptron.learn(numericDataSet.getAllDataPoints().get(i), d, 0.0001);
                complete = complete && d == y;
            }
            System.out.println("Minęło epok: " + counter++);
            if (counter > 10000)
                break;
        }
    }
}

File: src/perceptron/teacher.py
class Teacher:
    @staticmethod
    def teach(perceptron, data_set, class_value):
        complete = False
        counter = 1
        while not complete:
            complete = True
            for i in range(len(data_set.get_data_points())):
                d = 1 if class_value == data_set.get_decisions()[i] else 0
                y = perceptron.learn(data_set.get_data_points()[i], d)
                complete = complete and d == y
            print("Epochs passed:", counter)
            counter += 1
            if counter > 10000:
                break
